docs/adrs/0004-csi-proxy-and-configuration.md

```diff
@@ -0,0 +1,28 @@
+# 4. CSI Proxy and Configuration
+
+Date: 2021-11-22
+
+## Status
+
+Pending
+
+## Context
+
+To enable storage with Windows nodes, the [CSI Proxy](https://github.com/kubernetes-csi/csi-proxy) was created to allow CSI implementations to run as daemonsets on Windows nodes. Implementing support for installation and configuration of the CSI Proxy within Wins allows multiple projects to benefit from having the ability to have the CSI Proxy available, including both RKE and RKE2.
+
+## Decision
+
+After some discussion, it was decided that Wins would be responsible for downloading the specified version of CSI Proxy from the current upstream location based on the version that is specified. The ability to override the upstream URL will be provided. Once the binary is available Wins will be responsible for creating the Windows Service for CSI Proxy and ensuring that it is running. Wins will also have the ability to manage the lifecycle of the CSI Proxy service since configuration changes will need to be picked up from Wins. The proposed configuration is below:
+
+```yaml
+csi-proxy:
+  url: <location of the binaries>
+  version: <version>
+  managed: <bool>
```

Comment:
this is a bool to manage the service right? perhaps we only manage it if the `csi-proxy` config is here we just manage it any way... why would someone add this config and not expect wins to manage it?